src/proj_4D_api.c

```diff
@@ -793,6 +793,8 @@ PJ_INFO proj_info (void) {
     if (0 != info.version)
         return info;
 
+    pj_acquire_lock();
```

Comment:
the pj_acquire_lock() should be placed before the if(0 != info.version).
And there are missing pj_release_lock() in the early return code path